AEPServices/Sources/utility/ClassFinder.swift

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright 2023 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import Foundation
+
+public struct ClassFinder {
+    private static func allClasses() -> [AnyClass] {
+        let numberOfClasses = Int(objc_getClassList(nil, 0))
+        if numberOfClasses > 0 {
+            let classesPtr = UnsafeMutablePointer<AnyClass>.allocate(capacity: numberOfClasses)
+            let autoreleasingClasses = AutoreleasingUnsafeMutablePointer<AnyClass>(classesPtr)
+            let count = objc_getClassList(autoreleasingClasses, Int32(numberOfClasses))
+            assert(numberOfClasses == count)
+            defer { classesPtr.deallocate() }
+            let classes = (0 ..< numberOfClasses).map { classesPtr[$0] }
+            return classes
+        }
+        return []
+    }
+
+    public static func classes(conformToProtocol `protocol`: Protocol) -> [AnyClass] {
+        let classes = self.allClasses().filter { foundClass in
+            var anyClass: AnyClass? = foundClass
+            while let foundClass = anyClass {
+                if class_conformsToProtocol(foundClass, `protocol`) { return true }
```

Comment:
Can you run some tests with xcframeworks + optimization enabled and see if it still picks up the extension class?

Reply:
Tested with -Ofast level optimization and it still worked.